### DISCUSSION TITLE
disabling hooks

### DISCUSSION THREAD

**Discussion by ulric37** [flant/shell-operator#638] [Q&A]
Hello,
I'd like to use an environment variable to register a bash hook or not.
The idea is to use the same shell operator, on various environments, most hooks are common but some must be disabled in some contexts.
How can it be achieved ?
Thank you for your help !

**Comment by verdel** [comment]
@ulric37, you can disable hooks in the following way. In response to the hook being triggered with `--config`, we execute the following code and, depending on the value of the environment variable, return either a full configuration or the workaround described in this [issue](https://github.com/flant/shell-operator/issues/726):

```shell
if [ "$CUSTOM_LOGGING_FLOW_HOOK_ENABLED" == "true" ]; then
    cat <<EOF
{
  "configVersion":"v1",
  "kubernetes": [
    {
      "name": "Monitor the logging operator flow resources modifications",
      "apiVersion": "logging.banzaicloud.io/v1beta1",
      "kind": "flow",
      "executeHookOnEvent": [
        "Added",
        "Modified"
      ],
      "executeHookOnSynchronization": false
    }
  ]
}
EOF
  else
    cat <<EOF
{
  "configVersion":"v1",
  "settings": {
    "executionMinInterval": "1s",
    "executionBurst": 1
  }
}
EOF
  fi
}
```

**Reply by ulric37** [reply]
Thanks a lot !
